### PR TITLE
MBS-10989 extract inhouse report interface

### PR DESCRIPTION
**Pull request by dsvoronin**

### CODE REVIEW

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt [hunk 1]
```diff
@@ -32,6 +32,8 @@ public object InstrumentationPluginConfiguration {
          */
         public var useService: Boolean = false
 
+        public var useInMemoryReport: Boolean = false
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt]: Is it a temporary feature toggle or a long term config?
How about enum? 

report = [in-memory, avito-report]

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/suite/TestSuiteProvider.kt [hunk 2]
```diff
@@ -36,16 +40,28 @@ internal interface TestSuiteProvider {
...
+                val skippedTests = suite.skippedTests
+                    // do not report skip here, to prevent final test status rewrite (green from last run - ok)
+                    .filter { (_, verdict) ->
+                        verdict !is Excluded.BySignatures || verdict.source != Signatures.Source.PreviousRun
+                    }
+                    .map { (targetTest, verdict) ->
+                        targetTest.test to verdict.reason
+                    }
+
+                if (useInMemoryReport) {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/suite/TestSuiteProvider.kt:54]: Can we hide this decision inside a report?

It's easy to forget about checking toggle before sending\adding tests.

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/InstrumentationTestsAction.kt [hunk 3]
```diff
@@ -61,9 +61,7 @@ internal class InstrumentationTestsAction(
 
         val testsExecutionResults = scheduler.schedule()
 
-        finalizer.finalize(
-            testsExecutionResults = testsExecutionResults
-        )
+        finalizer.finalize(testsExecutionResults = testsExecutionResults)
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/InstrumentationTestsAction.kt:64]: I'd like to choose unambiguous and simple rules for suc formatting.
I see many changes in both ways in PRs. It makes harder to review.

This is out of scope, just thinking out loud.

File: subprojects/gradle/runner/report/src/main/kotlin/com/avito/android/runner/report/Report.kt [hunk 4]
```diff
@@ -1,27 +1,10 @@
 package com.avito.android.runner.report
 
-import com.avito.android.Result
 import com.avito.report.model.AndroidTest
-import com.avito.report.model.CrossDeviceSuite
-import com.avito.report.model.TestStaticData
 
-public interface Report : ReadReport {
+public interface Report : ReadReport, LegacyReport {
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/report/src/main/kotlin/com/avito/android/runner/report/Report.kt]: Can we make the opposite direction of inheritance?

`interface LegacyReport: Report`

If I need an acces to legacy API it'd better to make it explicitly.

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt]: temporary
report config already has it's own `testReport` function, which i don't want to interfere

Comment (eugene-krivobokov) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt]: Ok. We could express it explicitly in API:

```kotlin
experimental {
   report.inMemory = true
}
```

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/suite/TestSuiteProvider.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/suite/TestSuiteProvider.kt:54]: we can't; it uses separate interfaces depends on flag

File: subprojects/gradle/runner/report/src/main/kotlin/com/avito/android/runner/report/Report.kt (hunk 4 above)
Comment (dsvoronin) [reply subprojects/gradle/runner/report/src/main/kotlin/com/avito/android/runner/report/Report.kt]: i separated it completely now

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt]: fixed, take a look

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt (hunk 1 above)
Comment (eugene-krivobokov) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationPluginConfiguration.kt]: Thanks, looks self-explanatory
